crates/events/tests/web.rs

```diff
@@ -50,29 +49,29 @@ impl<A> Sender<A> {
     }
 }
 
-fn mpsc<A, F>(f: F) -> impl Future<Item = Vec<A>, Error = JsValue>
+async fn mpsc<A, F>(f: F) -> Vec<A>
 where
     F: FnOnce(Sender<A>),
 {
-    let (sender, receiver) = futures::sync::mpsc::unbounded();
+    let (sender, receiver) = futures::channel::mpsc::unbounded();
 
     f(Sender { sender });
 
     receiver
-        .then(|x| match x {
+        .map(|x| match x {
             Ok(a) => a,
             Err(_) => unreachable!(),
         })
-        .collect()
+        .collect().await
 }
 
 // ----------------------------------------------------------------------------
 // Tests begin here
 // ----------------------------------------------------------------------------
 
-#[wasm_bindgen_test(async)]
-fn new_with_options() -> impl Future<Item = (), Error = JsValue> {
-    mpsc(|sender| {
+#[wasm_bindgen_test]
+async fn new_with_options() {
+    let results = mpsc::<(), _>(|sender| {
```

Comment:
Is this type annotation needed? If so, I would prefer if it were `let results: Vec<()> = mpsc(|sender| {` instead.